### ISSUE TITLE
UndoManager double undo issue

### ISSUE THREAD

**Issue by CGNonofr** [yjs/yjs#241]
**Describe the bug**
The UndoManager seems to produce an incorrect internal state sometime.

**To Reproduce**
Run this code:
```javascript
const Y = require('yjs')

const doc = new Y.Doc()
const text = doc.getText()
text.insert(0, '1221')

const manager = new Y.UndoManager(text)

text.insert(2, '3')
text.insert(3, '3')

manager.undo()
manager.undo()

text.insert(2, '3')

console.log(text.toString())
```

or go to: https://repl.it/@LocMangeonjean/DimDangerousGuiltware#index.js

**Expected behavior**
the YText should contains '12321' instead of '31221'

**Screenshots**
![Capture du 2020-09-21 19-43-12](https://user-images.githubusercontent.com/23257479/93801717-b3a1fc00-fc42-11ea-9c83-5e458afb22ab.png)

**Environment Information**
 - Node v14.11.0
 - Yjs@13.3.2 without any extension

**Comment by MaximeCheramy** [comment]
I can reproduce the bug with yjs 13.3 and 13.4. But it works with yjs 13.2.0.

**Comment by CGNonofr** [comment]
After doing a `git bisect`, the breaking commit seems to be https://github.com/yjs/yjs/commit/6e3b708599ecfdd6c9c7f5af72debabbe74cc1af

**Comment by dmonad** [comment]
Thanks for digging into this! @MaximeCheramy @CGNonofr 

Makes sense, in than commit I introduced fast search marker - an experimental approach to find positions faster. Apparantly it doesn't work well with Undo/Redo. I will look into it!

Repl.it looks really nice for these kind of things. blitzstack is unfortunately extremely buggy.

**Comment by CGNonofr** [comment]
I tried to investigate a bit and managed to find a workaround until the bug is fixed:

```javascript
undoManager.on('stack-item-popped', () => {
      text._searchMarker = []
})
```

**Comment by dmonad** [comment]
Yeah, that should work :+1:

**Comment by dmonad** [comment]
This will be fixed in 13.4.0.

@CGNonofr You might run into issues with your solution because if `_searchMarker=null` then the feature should be disabled. Searchmarkers are currently only supported in YText and YArray. Setting _searchMarker=[] in, for example, XmlElements will lead to issues. So I encourage you to upgrade ASAP.

**Comment by CGNonofr** [comment]
Good to know! btw I only do it on a YText so I guess it's fine

Anyway I'll still update it as soon as it's published of course

Thanks :+1:
